src/string_with_vars.ml

```diff
@@ -136,74 +143,87 @@ module Partial = struct
     | Unexpanded of t
 end
 
-let invalid_multivalue syntax ~var t x =
-  Loc.fail t.loc "Variable %s expands to %d values, \
+let invalid_multivalue (v : var) x =
+  Loc.fail v.loc "Variable %s expands to %d values, \
                   however a single value is expected here. \
                   Please quote this atom."
-    (string_of_var syntax var) (List.length x)
-
-let partial_expand t ~mode ~dir ~f =
-  let commit_text acc_text acc =
-    let s = concat_rev acc_text in
-    if s = "" then acc else Text s :: acc
-  in
-  let rec loop acc_text acc items =
-    match items with
-    | [] ->
-      begin match acc with
-      | [] -> Partial.Expanded (Mode.string mode (concat_rev acc_text))
-      | _  -> Unexpanded { t with items = List.rev (commit_text acc_text acc) }
-      end
-    | Text s :: items -> loop (s :: acc_text) acc items
-    | Var (syntax, var) as it :: items ->
-      begin match f syntax t.loc var with
-      | Some ([] | _::_::_ as e) when not t.quoted ->
-        invalid_multivalue syntax ~var t e
-      | Some t ->
-        loop (Value.L.concat ~dir t :: acc_text) acc items
-      | None -> loop [] (it :: commit_text acc_text acc) items
+    (string_of_var v) (List.length x)
+
+(* let rec rev_append acc = function
```

Comment:
You might want to delete this part (if it's not for the remaining TODO parts of course).